plugins/chat/spec/services/chat_message_destroyer_spec.rb

```diff
@@ -0,0 +1,50 @@
+# frozen_string_literal: true
+
+RSpec.describe ChatMessageDestroyer do
+  describe "#destroy_in_batches" do
+    fab!(:chat_message) { Fabricate(:chat_message) }
+    fab!(:user_1) { Fabricate(:user) }
+
+    it "resets last_read_message_id from memberships" do
+      membership =
+        UserChatChannelMembership.create!(
+          user: user_1,
+          chat_channel: chat_message.chat_channel,
+          last_read_message: chat_message,
+          following: true,
+          desktop_notification_level: 2,
+          mobile_notification_level: 2,
+        )
+
+      described_class.new.destroy_in_batches(ChatMessage.where(id: chat_message.id))
+
+      expect(membership.reload.last_read_message_id).to be_nil
+    end
+
+    it "deletes flags associated to deleted chat messages" do
+      guardian = Guardian.new(Discourse.system_user)
+      Chat::ChatReviewQueue.new.flag_message(
+        chat_message,
+        guardian,
+        ReviewableScore.types[:off_topic],
+      )
+
+      reviewable = ReviewableChatMessage.last
+      expect(reviewable).to be_present
+
+      described_class.new.destroy_in_batches(ChatMessage.where(id: chat_message.id))
+
+      expect { chat_message.reload }.to raise_exception(ActiveRecord::RecordNotFound)
+      expect { reviewable.reload }.to raise_exception(ActiveRecord::RecordNotFound)
+    end
+
+    it "doesn't delete other messages" do
+      message_2 = Fabricate(:chat_message, chat_channel: chat_message.chat_channel)
+
+      described_class.new.destroy_in_batches(ChatMessage.where(id: chat_message.id))
```

Comment:
There's a little bit of naming inconsistency here, sometimes we say `chat_message`, sometimes `message`.

I would recommend to go with this: `fab!(:message_1) {}` and `fab!(:message_2) {}`

It's the less verbose, super consistent and still very clear.